homeassistant/components/decora_wifi/common.py

```diff
@@ -0,0 +1,156 @@
+        """Update the device library from the API."""
+
+        try:
+            # Gather all the available devices into the iot_switches dictionary...
+            perms = self._session.user.get_residential_permissions()
+
+            for permission in perms:
+                if permission.residentialAccountId is not None:
+                    acct = ResidentialAccount(
+                        self._session, permission.residentialAccountId
+                    )
+                    residences = acct.get_residences()
+                    for res in residences:
+                        switches = res.get_iot_switches()
+                        for switch in switches:
+                            # Add the switch to the appropriate list in the iot_switches dictionary.
+                            platform = DecoraWifiPlatform.classifydevice(switch)
+                            self._iot_switches[platform].append(switch)
+                elif permission.residenceId is not None:
+                    residence = Residence(self._session, permission.residenceId)
+                    switches = residence.get_iot_switches()
+                    for switch in switches:
+                        # Add the switch to the appropriate list in the iot_switches dictionary.
+                        platform = DecoraWifiPlatform.classifydevice(switch)
+                        self._iot_switches[platform].append(switch)
+        except ValueError as exc:
+            raise DecoraWifiCommFailed from exc
+
+    def reauth(self):
+        """Reauthenticate this object's session."""
+        self.apilogout()
+        self._session = DecoraWiFiSession()
+        self._apilogin()
+
+    def refresh_devices(self):
+        """Refresh this object's devices."""
+        self._iot_switches: dict[str, IotSwitch] = {
+            platform: [] for platform in PLATFORMS
+        }
+        self._apigetdevices()
+
+    @staticmethod
+    async def async_setup_decora_wifi(hass: HomeAssistant, email: str, password: str):
+        """Set up a decora wifi session."""
+
+        def setupplatform() -> DecoraWifiPlatform:
+            return DecoraWifiPlatform(email, password)
+
+        return await hass.async_add_executor_job(setupplatform)
+
+    @staticmethod
+    def classifydevice(dev):
+        """Classify devices by platform."""
+        # The light platform is the only one currently implemented in the integration.
+        return LIGHT_DOMAIN
+
+
+decorawifisessions: dict[str, DecoraWifiPlatform] = {}
+
+
+class DecoraWifiEntity(Entity):
+    """Initiate Decora Wifi Base Class."""
+
+    def __init__(self, device):
+        """Initialize Decora Wifi device base class."""
+        self._switch = device
+        self._unique_id = device.mac
+
+    async def async_added_to_hass(self):
+        """Run on addition of device to hass."""
+        self.async_on_remove(
+            async_dispatcher_connect(self.hass, DOMAIN, self.async_write_ha_state)
```

Comment:
If the user has multiple config entries this will fire when an unrelated one updates

Reply:
Researching what needs to be done with this now. I included it because I was imitating another integration but it may not be needed at all.